ipatests/test_integration/test_epn.py

```diff
@@ -494,3 +497,78 @@ def test_EPN_delay_config(self, cleanupmail):
         self.master.put_file_contents('/etc/ipa/epn.conf', epn_conf)
         result = tasks.ipa_epn(self.master, raiseonerr=False)
         assert "smtp_delay cannot be less than zero" in result.stderr_text
+
+    def generate_user_batch(self):
+        """
+        Prepare large (10k) batch of users with various attributes, namely
+        without passwords, gradually expiring passwords, unicode names and
+        unicode e-mails
+        :return: None
+        """
+        # number of groups: 5 (one is split in two)
+        # describe list format in deque
+        USERS_IN_GROUP = 2000
+        # inactive users without passwords
```

Comment:
I am thinking about merging all the deques into one, which would simplify removal and checking should be unaffected, as both username and expiry date are stored

Reply:
I merged it into one big deque. But worry not, deques are nice and effective.